### PR TITLE
fix: [CDS-68796]: Retry field for jira approval and service now approval

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by litvinovstas — APPROVED** [review]

**Review by PratyushGar — DISMISSED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger uts

**Review by vikrant-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx [hunk 1]
```diff
@@ -395,7 +406,22 @@ function ServiceNowApprovalStepMode(
   })
   return (
     <Formik<ServiceNowApprovalData>
-      onSubmit={values => onUpdate?.(values)}
+      onSubmit={values => {
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx:409]: Lets try to reduce the duplication of code whenever possible - please extract this out in a common util (isRetryIntervalGreaterThanTimeout) which takes in formikValues(formik.current.values) and returns a boolean (retryInterval > timeout)

And then lets use it at both places

File: src/modules/70-pipeline/components/PipelineSteps/Steps/JiraApproval/JiraApprovalStepMode.tsx [hunk 2]
```diff
@@ -524,6 +539,19 @@ function JiraApprovalStepMode(props: JiraApprovalStepModeProps, formikRef: StepF
   return (
     <Formik<JiraApprovalData>
       onSubmit={values => {
+        const timeoutString = (formikRef as React.MutableRefObject<FormikProps<JiraApprovalData>>)?.current?.values
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/JiraApproval/JiraApprovalStepMode.tsx:542]: https://github.com/harness/harness-core-ui/pull/17088/files#r1292712473

Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/JiraApproval/JiraApprovalStepMode.tsx]: addressed

File: src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx (hunk 1 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/ServiceNowApproval/ServiceNowApprovalStepMode.tsx:408]: addressed

**Review by PratyushGar — APPROVED** [review]

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Pull request merged**